common/beerocks/bwl/nl80211/sta_wlan_hal_nl80211.cpp

```diff
@@ -41,12 +63,136 @@ bool sta_wlan_hal_nl80211::connect(const std::string &ssid, const std::string &p
                                    bool mem_only_psk, const std::string &bssid, uint8_t channel,
                                    bool hidden_ssid)
 {
+    LOG(DEBUG) << "Connect interface " << get_iface_name() << " to SSID = " << ssid
+               << ", BSSID = " << bssid << ", Channel = " << int(channel)
+               << ", Sec = " << nl80211_security_val(sec) << ", mem_only_psk=" << int(mem_only_psk);
+
+    if (ssid.empty() || bssid.empty() || sec == WiFiSec::Invalid) {
+        LOG(ERROR) << "Invalid params!";
+        return false;
+    }
+
+    // First disconnect (or do nothing if not connected)
+    if (!disconnect()) {
+        LOG(WARNING) << "Failed disconnecting before connecting to the new BSSID";
+        return false;
+    }
+
+    // Add a new network
+    auto network_id = add_network();
+    if (network_id < 0) {
+        LOG(ERROR) << "Failed (" << network_id
+                   << ") adding new network to interface: " << get_iface_name();
+        return false;
+    }
+
+    auto freq = son::wireless_utils::channel_to_freq(int(channel));
+
+    // Update network parameters
+    if (!set_network_params(network_id, ssid, bssid, sec, mem_only_psk, pass, hidden_ssid, freq)) {
+        LOG(ERROR) << "Failed setting network id = " << network_id
+                   << " on interface: " << get_iface_name();
+        return false;
+    }
+
+    // Enable the network
+    if (!enable_network(network_id)) {
+        LOG(ERROR) << "Failed enabling network id = " << network_id
+                   << " on interface: " << get_iface_name();
+        return false;
+    }
+
+    LOG(ERROR) << "Network with id " << network_id << " has been added and enabled on interface "
```

Comment:
```suggestion
    LOG(DEBUG) << "Network with id " << network_id << " has been added and enabled on interface "
```
this is not an error